jwst/resample/resample.py

```diff
@@ -188,48 +187,41 @@ def resample_many_to_one(self):
     def resample_variance_array(self, name, output_model):
         """Resample variance arrays from self.input_models to the output_model
 
-        Loop through self.input_models and resample the `name` variance array
-        to the same name in output_model.  This modifies output_model in-place.
+        Resample the `name` variance array to the same name in output_model,
+        using a cummulative sum.
+
+        This modifies output_model in-place.
         """
-        outwht = np.zeros_like(output_model.data)
-        outcon = np.zeros_like(output_model.con)
         output_wcs = output_model.meta.wcs
-        output_arrays = []
+        inverse_variance_sum = np.zeros_like(output_model.data)
 
         log.info(f"Resampling {name}")
         for model in self.input_models:
-            # Unity weight but ignore pixels that are NON_SCIENCE or REFERENCE_PIXEL
+            variance = getattr(model, name)
+            # Make input weight map of unity where there is science data
             inwht = resample_utils.build_driz_weight(model, weight_type=None,
                                                      good_bits="~NON_SCIENCE+REFERENCE_PIXEL")
-            input_array = getattr(model, name)
-            output_array = np.zeros_like(output_model.data)
 
-            # Resample the variance array
-            self.drizzle_arrays(input_array, inwht, model.meta.wcs,
-                                output_wcs, output_array, outwht, outcon,
+            resampled_variance = np.zeros_like(output_model.data)
+            outwht = np.zeros_like(output_model.data)
+            outcon = np.zeros_like(output_model.con)
+
+            # Resample the variance array.  Use fillval=np.inf so that when we
+            # take the reciprocal for summing, it is zero where there is zero weight
+            self.drizzle_arrays(variance, inwht, model.meta.wcs,
+                                output_wcs, resampled_variance, outwht, outcon,
                                 pixfrac=self.pixfrac, kernel=self.kernel,
-                                fillval=np.nan)
-            output_arrays.append(output_array)
+                                fillval=np.inf)
 
-        # Stack and coadd as inverse variance if there any to stack
-        if len(output_arrays) > 1:
-            stacked = np.stack(output_arrays)
+            # Add the inverse of the resampled variance to a running sum
             with np.errstate(divide="ignore"):
-                inv_var = np.reciprocal(stacked)
-            inv_var[~np.isfinite(inv_var)] = np.nan
-
-            with warnings.catch_warnings():
-                warnings.filterwarnings(action="ignore",
-                                        message="Mean of empty slice",
-                                        category=RuntimeWarning)
-                inv_var_mean = np.nanmean(inv_var, axis=0)
```

Comment:
So this line is a bug in the original code.  Instead of of summing the inverse variances, I was taking the mean along the stack with no reference to number of images in the overlapping regions.  Which means the final error array was overestimated.  Ack.

So in our regression test for NIRCam imaging with 3 dithered pointings.  The error array in the resampled image was computed as the following for regions with 1, 2 and 3 overlapping pointings, respectively:

```
In [10]: truth.err[2250:2300,3400:3450].mean()
Out[10]: 0.024356956

In [12]: truth.err[2100:2150,3400:3450].mean()
Out[12]: 0.019771107

In [16]: truth.err[1850:1900,3400:3450].mean()
Out[16]: 0.017140672
```

This PR correctly sums the inverse variances, so that for similar variances, the combined variance of N images shows a reduction in sigma by `sqrt(N)` where there are N overlapping resampled images.

I.e. for the same regions above:

```
In [9]: result.err[2250:2300,3400:3450].mean()
Out[9]: 0.024356956

In [11]: result.err[2100:2150,3400:3450].mean()
Out[11]: 0.01714876

In [15]: result.err[1850:1900,3400:3450].mean()
Out[15]: 0.01403387
```

And it scales as ~`1 / sqrt(N)` as expected:

```
In [17]: 0.024356956 / np.sqrt(2)
Out[17]: 0.017222968756662363

In [18]: 0.024356956 / np.sqrt(3)
Out[18]: 0.014062495103239871
```

Reply:
I've put the above into a unit test.